### PR TITLE
Issue #SB-30154 feat: DIAL Link API refactoring.

### PR DESCRIPTION
**Pull request by Jayaprakash8887**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-30154" title="SB-30154" target="_blank">TaskSB-30154</a>  Part 2: Collection Dialcode Link API Refactor LP to KP
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Ran Unit Tests

**Test Configuration**:
* Software versions: Java 11, scala-2.11, play-2.7.2
* Hardware versions: 2 CPU/ 4GB RAM

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala [hunk 1]
```diff
@@ -59,14 +59,20 @@ object DIALManager {
...
+			requestList.foreach(req => {
+				val contents: List[String] = req(DIALConstants.IDENTIFIER)
+				val dialcodes: List[String] = req(DIALConstants.DIALCODE)
+				validateReqStructure(dialcodes, contents)
+				contents.foreach(id => reqMap += (id -> dialcodes))
+			})
+		} catch {
+			case e: Exception => throw new ClientException(DIALErrors.ERR_DIALCODE_CONTENT_LINK_FIELDS_MISSING, DIALErrors.ERR_DIALCODE_CONTENT_LINK_FIELDS_MISSING_MSG)
+		}
+		TelemetryManager.info("DIALManager::validateAndGetRequestMap:: requestList: " + requestList)
```
Comment (vinukumar-vs) [inline content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala:72]: Log any api failures/exceptions as TelemetryManager.warn message for ease of discovery

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: content-api/content-actors/src/test/resources/application.conf [hunk 2]
```diff
@@ -495,7 +495,7 @@ dial_service {
   api {
     base_url = "https://qa.ekstep.in/api"
     auth_key = "auth_key"
-    search = "/dialcode/v1/search"
+    search = "/dialcode/v3/search"
```
[the remark this answers is not in the document]
Comment (Jayaprakash8887) [reply content-api/content-actors/src/test/resources/application.conf:498]: No. The backend api endpoint was always v3. I had committed it worng as v1

File: content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala (hunk 1 above)
Comment (Jayaprakash8887) [reply content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala:72]: ClientException gets logged. This is enough for finding out the issue.

Comment (Jayaprakash8887) [reply content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala:68]: Added Exceptions log themselves. No need to log separately. I will remove info logs and recommit once QA testing is completed. They have been added for debugging purposes only.

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**